internal/git/config.go

```diff
@@ -9,16 +9,33 @@ import (
 	"github.com/pkg/errors"
 )
 
+//go:generate moq -out config_mock.go . GitConfigAPI
+
+// GitConfigAPI is an interface to interact with a git config system
+// this makes it possible to extract information from the repository
+// or the local user
+type GitConfigAPI interface {
+	CommitterDetails() (name string, email string, err error)
+}
+
+type LocalGitConfigAPI struct {
+}
+
+// NewLocalGitConfigAPI provide a GitConfigAPI for a local repository
+func NewLocalGitConfigAPI() GitConfigAPI {
```

Comment:
Same as above. We normally want to accept interfaces and return concrete types. Is this intentionally doing something else?

Reply:
This is a pattern often used in other languages. However, it does mean that a struct can only implement a single interface. I will move the interface closer to the user, and as such this will change to the concrete type instead.

Reply:
I've moved it closer to the caller, this does introduce some extra boiler plate for the interfaces, as it is now declared where it is used.